kiwi/container/setup/appx.py

```diff
@@ -0,0 +1,155 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+import os
+import platform
+from lxml import etree
+from xml.dom import minidom
+
+# project
+from kiwi.container.setup.base import ContainerSetupBase
+from kiwi.exceptions import KiwiContainerSetupError
+
+
+class ContainerSetupAppx(ContainerSetupBase):
+    """
+    WSL/Appx container setup
+    """
+    def setup(self):
+        """
+        Setup container metadata
+        """
+        self.appx_at_microsoft = 'http://schemas.microsoft.com/appx'
+        self.appx_manifest_file = os.sep.join(
+            [self.custom_args.get('metadata_path'), 'AppxManifest.xml']
+        )
+        self.appx_namespace_map = dict(
+            wsl='{0}/manifest/foundation/windows10'.format(
+                self.appx_at_microsoft
+            ),
+            uap='{0}/manifest/uap/windows10'.format(
+                self.appx_at_microsoft
+            ),
+            desktop='{0}/manifest/desktop/windows10'.format(
+                self.appx_at_microsoft
+            ),
+            uap3='{0}/manifest/uap/windows10/3'.format(
+                self.appx_at_microsoft
+            ),
+            mp='{0}/2014/phone/manifest'.format(
+                self.appx_at_microsoft
+            )
+        )
+        if not os.path.exists(self.appx_manifest_file):
+            raise KiwiContainerSetupError(
+                'Appx Manifest not found at: {0}'.format(
+                    self.appx_manifest_file
+                )
+            )
+        try:
+            self.appx_manifest = etree.parse(
+                self.appx_manifest_file
+            )
+            self._update_identity()
+            self._update_desktop_visual()
+            self._update_application_metadata()
+            xml_data_unformatted = etree.tostring(
+                self.appx_manifest.getroot()
+            )
+            xml_data_domtree = minidom.parseString(xml_data_unformatted)
+            xml_data_raw = xml_data_domtree.toprettyxml(
+                indent='    ', encoding="utf-8"
+            )
+            with open(self.appx_manifest_file, 'wb') as appx_manifest:
+                for line in xml_data_raw.split(b'\n'):
+                    if line.split():
+                        appx_manifest.write(line + b'\n')
+        except Exception as issue:
+            raise KiwiContainerSetupError(
+                'Failed to update Appx Manifest: {0}'.format(issue)
+            )
+
+    def _update_identity(self):
+        identities = self._get_xpath('//wsl:Identity')
+        history = self.custom_args.get('history') or {}
+        for identity in identities:
+            identity.set('ProcessorArchitecture', platform.machine())
+            identity.set('Version', self.custom_args.get('image_version'))
+            if history.get('created_by'):
+                identity.set('Publisher', history.get('created_by'))
```

Comment:
`Publisher` has to equal what is set in the project certificate, so kiwi should either find that out by itself or not modify this.

Reply:
I suggest to set the values below `Properties` instead.

Reply:
ok so we don't set Publisher.  I don't see a generic way for kiwi to do this

> I suggest to set the values below Properties instead.

This is done, we set:

DisplayName
PublisherDisplayName

from Properties. Do we need more ?

Reply:
in the visuals update